pkg/activator/net/revision_backends.go

```diff
@@ -253,7 +263,7 @@ func (rw *revisionWatcher) probePodIPs(ready, notReady sets.String) (succeeded s
 		dest := dest // Standard Go concurrency pattern.
 		probeGroup.Go(func() error {
 			ok, notMesh, err := rw.probe(egCtx, dest)
-			if ok {
+			if ok && (ready.Has(dest) || rw.enableProbeOptimisation) {
```

Comment:
I know that Go prefers [fewer parentheses](https://golang.org/doc/effective_go#formatting), but I still had to stop and think here for a split second (this is a Scrooge-level nit, albeit of the McDuck variety rather than Ebenezer...)

Reply:
I'm probably missing obvious, but I don't think there's a way to put any more parentheses on this line?

Reply:
Never mind me... I wasn't looking at the right commit